lib/optimize.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+var _ = require('underscore');
+
+/* global JuttleAdapterAPI */
+var JuttleMoment = JuttleAdapterAPI.types.JuttleMoment;
+var reducerDefaultValue = JuttleAdapterAPI.runtime.reducerDefaultValue;
+
+let logger = JuttleAdapterAPI.getLogger('ga-optimize');
+
+function _getFieldName(node) {
+    if (node.type === 'UnaryExpression' && node.operator === '*' && node.argument.d) {
+        return node.argument.value;
+
+    } else { // node.type === 'Field'
```

Comment:
I originally did that. But then my code coverage was < 100% so I restructured this way.

Admittedly that's a bit cowardly, but so is adding a branch to throw an Error (that I can't ever trip) and then reducing the branch coverage to something like 99.X%. So I went this route instead.